### PR TITLE
Compiler constants refactor

### PR DESCRIPTION
**Pull request by ElijahAhianyo**
Moved compiled constants into env.json file in the .web folder

### All Submissions:

- [ ] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [ ] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

fixes [#1012](https://github.com/pynecone-io/pynecone/issues/1012)

### CODE REVIEW

File: pynecone/.templates/jinja/web/pages/base_page.js.jinja2 [hunk 1]
```diff
@@ -4,6 +4,7 @@
 {% for module in imports%}
   {{- utils.get_import(module) }}
 {% endfor %}
+import env from  "{{env_file}}"
```
Comment (picklelo) [inline pynecone/.templates/jinja/web/pages/base_page.js.jinja2:7]: I think we can get rid of this - the constants should only be needed in the `state.js` file now

File: pynecone/components/layout/foreach.py [hunk 2]
```diff
@@ -13,15 +13,13 @@ class Foreach(Component):
     """A component that takes in an iterable and a render function and renders a list of components."""
 
     # The iterable to create components from.
-    iterable: Var[Union[List, Dict, Tuple, Set]]
+    iterable: Var[Iterable]
```
Comment (picklelo) [inline pynecone/components/layout/foreach.py:16]: nice!

**Review by picklelo — COMMENTED** [review]
Thanks for this change!

I think we don't need the `.env` inside the component files anymore. We can import it in `state.js` instead so that file will have all the constants. Currently we are passing in the constants to the state.js files which is why we needed it here, but we can remove that now.

**Comment by ElijahAhianyo** [comment]
> Thanks for this change!
> 
> I think we don't need the `.env` inside the component files anymore. We can import it in `state.js` instead so that file will have all the constants. Currently we are passing in the constants to the state.js files which is why we needed it here, but we can remove that now.

Yeah that makes sense. Will update this accordingly

**Review by picklelo — APPROVED** [review]
awesome, thanks for this update!

**Pull request merged**